packages/eslint-plugin/tests/rules/unbound-method.test.ts

```diff
@@ -58,6 +58,23 @@ ruleTester.run('unbound-method', rule, {
     '[5.2, 7.1, 3.6].map(Math.floor);',
     'const x = console.log;',
     'const x = Object.defineProperty;',
+    `
+      const o = {
+        f: function (this: void) {},
+      };
+      const f = o.f;
+    `,
+    `
```

Comment:
the `window` test are not related to the bug, just things i came up with while trying to figure out how to satisfy codecov.